### DISCUSSION TITLE
Need for better webpage retrieval

### DISCUSSION THREAD

**Discussion by mayurdb** [ScrapeGraphAI/Scrapegraph-ai#188] [General]
Currently, FetchNode uses the AsyncChromiumLoader to load the page and the returned content is used further.

I think it would be good if we invested more effort into getting more structured and detailed webpage content:
I mean getting a dict out of the page that includes
1. Static content
2. Forms
3. Image URLs
4. Links
5. Getting dynamically loaded content on the page (I mean on scroll etc)

**Comment by vvkishere** [comment]
I think there should be a node that just saves the data scraped either to a vector store or to a local directory. Sometimes the need for the scraping is not just to generate answers.

**Comment by VinciGit00** [comment]
There is the rag node
